haskell/src/Stablecoin/Client/Parser.hs

```diff
@@ -0,0 +1,683 @@
+      mkCommandParser
+        "get-pending-contract-owner"
+        (pure CmdGetPendingContractOwner)
+        "Check if there's a pending contract owner"
+
+    getMasterMinterCmd :: Opt.Mod Opt.CommandFields ClientArgsRaw
+    getMasterMinterCmd =
+      mkCommandParser
+        "get-master-minter"
+        (pure CmdGetMasterMinter)
+        "Get the master minter's address"
+
+    getPauserCmd :: Opt.Mod Opt.CommandFields ClientArgsRaw
+    getPauserCmd =
+      mkCommandParser
+        "get-pauser"
+        (pure CmdGetPauser)
+        "Get the pauser's address"
+
+    getTransferlistCmd :: Opt.Mod Opt.CommandFields ClientArgsRaw
+    getTransferlistCmd =
+      mkCommandParser
+        "get-transferlist"
+        (pure CmdGetTransferlist)
+        "Get the transferlist's address"
+
+    getTotalSupplyCmd :: Opt.Mod Opt.CommandFields ClientArgsRaw
+    getTotalSupplyCmd =
+      mkCommandParser
+        "get-total-supply"
+        (pure CmdGetTotalSupply)
+        "Get the total supply of tokens"
+
+    getMintingAllowanceCmd :: Opt.Mod Opt.CommandFields ClientArgsRaw
+    getMintingAllowanceCmd =
+      mkCommandParser
+        "get-minting-allowance"
+        (CmdGetMintingAllowance <$> getMintingAllowanceOptions)
+        "Get the minting allowance for a minter"
+
+    getMintingAllowanceOptions :: Opt.Parser GetMintingAllowanceOptions
+    getMintingAllowanceOptions = do
+      gmaoMinter <-
+        addressOrAliasOption
+          Nothing
+          (#name .! "minter")
+          (#help .! "Address or alias of the minter whose allowance will be retrieved")
+      pure $ GetMintingAllowanceOptions {..}
+
+    getTokenMetadataCmd :: Opt.Mod Opt.CommandFields ClientArgsRaw
+    getTokenMetadataCmd =
+      mkCommandParser
+        "get-token-metadata"
+        (pure CmdGetTokenMetadata)
+        "Get the token metadata"
+
+stablecoinClientInfo :: Opt.ParserInfo ClientArgs
+stablecoinClientInfo =
+  Opt.info (Opt.helper <*> versionOption <*> clientArgsParser) $
+  mconcat
+    [ Opt.fullDesc
+    , Opt.progDesc $
+        "Tezos Stablecoin Client: Command line interface for interacting " <>
+        "with the Tezos Stablecoin contract"
+    , Opt.header "Stablecoin Client"
+    , Opt.footerDoc $ usageDoc
+    ]
+
+  where
+    versionOption =
+      Opt.infoOption
+        ("stablecoin-" <> showVersion version)
+        (Opt.long "version" <> Opt.help "Show version.")
+
+    usageDoc :: Maybe Doc
+    usageDoc =
+      Just $ mconcat
+      [ "You can use help for specific COMMAND", linebreak
+      , "EXAMPLE:", linebreak
+      , "  stablecoin-client mint --help", linebreak
+      , "USAGE EXAMPLE:", linebreak
+      , "  stablecoin-client mint --to tz1U1h1YzBJixXmaTgpwDpZnbrYHX3fMSpvb --amount 100500", linebreak
+      , linebreak
+      , "  This command will produce the given amounts of tokens to", linebreak
+      , "  the wallets associated with the given address.", linebreak
+      ]
+
+-- TODO: add this orphan instance to morley
```

Comment:
Maybe let's add it in https://gitlab.com/morley-framework/morley/-/merge_requests/532?

Reply:
Done